usaspending_api/etl/management/commands/reindex_elastic.py

```diff
@@ -0,0 +1,157 @@
+        source_indices = sorted(source_indices, key=lambda x: x.split('-')[-1])
+
+        dest_indices = [i.replace(self.source_prefix, self.dest_prefix) for i in source_indices]
+        self.reindex_dict = dict(zip(source_indices, dest_indices))
+        count = self.get_total_count(index_name)
+        logging.warn('Reindexing a total of {} documents'.format(count))
+
+    def get_total_count(self, index_name):
+        try:
+            response = ES_CLIENT.search(index=index_name)
+            return response["hits"]["total"]
+        except TransportError as e:
+            logging.error('[CONNECTION TIMEOUT] with ElasticSearch cluster: {e}'.
+                          format(e=str(e)))
+            return -1
+
+    def verify(self, source, target):
+        difference = self.get_total_count(source) - self.get_total_count(target)
+        if difference != 0:
+            logging.warn('Still have {} documents to reindex'.format(difference))
+        return difference, difference/INGEST_RATE
+
+    def controller(self):
+        logging.warn('Reindexing the following \n')
+        pprint(self.reindex_dict)
+        print('\n\n')
+        for source, dest in self.reindex_dict.items():
+            if ES_CLIENT.indices.exists(dest):
+                if not self.wipe:
+                    logging.warn('{} exists... passing ... '.format(dest))
+                else:
+                    logging.warn('{} exists... deleting... {}'.format(dest))
+                    ES_CLIENT.indices.delete(dest)
+            if not ES_CLIENT.indices.exists(dest):
+                self.size, self.sleep_time = self.verify(source, dest)
+                logging.warn('Reindexing {} -- {} documents'.format(source, self.size))
+                try:
+                    self.single_reindex(source, dest)
+                    logging.info(source)
+                    ES_CLIENT.indices.flush('*')
+                except (ConnectionTimeout) as e:
+                    logging.error('[CONNECTION TIMEOUT] with ElasticSearch cluster: {e}'.
+                                  format(e=str(e)))
+                    for attempt in range(MAX_RETRIES):
+                        self.size, self.sleep_time = self.verify(source, dest)
+                        if self.sleep_time > 0:
+                            logging.warn('---------- Sleep_time of index >>> {} ------------------'
+                                         .format(str(self.sleep_time)))
+                            time.sleep(self.sleep_time)
+        print('\n\n')
+        index_name = self.dest_prefix + '*'
+        ES_CLIENT.indices.flush('*')
+        time.sleep(5)
+        count = self.get_total_count(index_name)
+        logging.warn('Reindexed a total of {} documents'.format(count))
+        logging.warn('****   FINISHED REINDEXING  ****')
+
+        for source, dest in self.reindex_dict.items():
+            transaction_category = next((s for s in INDICES_SUB_NAMES if dest.find(s) != -1), None)
```

Comment:
Based on the script title I didn't expect it to be tightly coupled to the current index naming convention. Is there an expectation to run this on a regular bases or only when necessary?

Reply:
It is coupled with current. We need to discuss this vs reloading. We shouldn't have to run this on a regular basis.

Reply:
Since this isn't planned for regular execution, I would argue this script will be more helpful if it isn't tied to the current award category descriptions. That way it could be used for any reindex job